<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Enhanced Influence Model | MATH 479 Network Science Project</title>
  <meta name="description" content="Chapter 5 Enhanced Influence Model | MATH 479 Network Science Project" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Enhanced Influence Model | MATH 479 Network Science Project" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="zuofuhuang/NetworkProject" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Enhanced Influence Model | MATH 479 Network Science Project" />
  
  
  

<meta name="author" content="Zhaoheng Li and Zuofu Huang, Fall 2020 Macalester College" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="network-construction.html"/>
<link rel="next" href="discussion-and-future-work.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="initial-steps.html"><a href="initial-steps.html"><i class="fa fa-check"></i><b>2</b> Initial Steps</a><ul>
<li class="chapter" data-level="2.1" data-path="initial-steps.html"><a href="initial-steps.html#genetics-background"><i class="fa fa-check"></i><b>2.1</b> Genetics Background</a></li>
<li class="chapter" data-level="2.2" data-path="initial-steps.html"><a href="initial-steps.html#simulate-original-data"><i class="fa fa-check"></i><b>2.2</b> Simulate original data</a></li>
<li class="chapter" data-level="2.3" data-path="initial-steps.html"><a href="initial-steps.html#sample-mutations-for-patients-at-snp-locations"><i class="fa fa-check"></i><b>2.3</b> Sample mutations for patients at SNP locations</a></li>
<li class="chapter" data-level="2.4" data-path="initial-steps.html"><a href="initial-steps.html#sample-mutations-for-non-patients-at-snp-locations"><i class="fa fa-check"></i><b>2.4</b> Sample mutations for non-patients at SNP locations</a></li>
<li class="chapter" data-level="2.5" data-path="initial-steps.html"><a href="initial-steps.html#variants-at-non-snp-locations"><i class="fa fa-check"></i><b>2.5</b> Variants at non-SNP locations</a></li>
<li class="chapter" data-level="2.6" data-path="initial-steps.html"><a href="initial-steps.html#screening"><i class="fa fa-check"></i><b>2.6</b> Screening</a></li>
<li class="chapter" data-level="2.7" data-path="initial-steps.html"><a href="initial-steps.html#marginal-regression"><i class="fa fa-check"></i><b>2.7</b> Marginal Regression</a></li>
<li class="chapter" data-level="2.8" data-path="initial-steps.html"><a href="initial-steps.html#discussion"><i class="fa fa-check"></i><b>2.8</b> Discussion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="progress-and-revision.html"><a href="progress-and-revision.html"><i class="fa fa-check"></i><b>3</b> Progress and Revision</a><ul>
<li class="chapter" data-level="3.1" data-path="progress-and-revision.html"><a href="progress-and-revision.html#rationale"><i class="fa fa-check"></i><b>3.1</b> Rationale</a></li>
<li class="chapter" data-level="3.2" data-path="progress-and-revision.html"><a href="progress-and-revision.html#updated-code"><i class="fa fa-check"></i><b>3.2</b> Updated Code</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="network-construction.html"><a href="network-construction.html"><i class="fa fa-check"></i><b>4</b> Network Construction</a><ul>
<li class="chapter" data-level="4.1" data-path="network-construction.html"><a href="network-construction.html#network-construction-process"><i class="fa fa-check"></i><b>4.1</b> Network Construction Process</a></li>
<li class="chapter" data-level="4.2" data-path="network-construction.html"><a href="network-construction.html#analysis"><i class="fa fa-check"></i><b>4.2</b> Analysis</a></li>
<li class="chapter" data-level="4.3" data-path="network-construction.html"><a href="network-construction.html#centrality-measures-report"><i class="fa fa-check"></i><b>4.3</b> Centrality Measures Report</a></li>
<li class="chapter" data-level="4.4" data-path="network-construction.html"><a href="network-construction.html#preliminary-conclusions"><i class="fa fa-check"></i><b>4.4</b> Preliminary Conclusions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="enhanced-influence-model.html"><a href="enhanced-influence-model.html"><i class="fa fa-check"></i><b>5</b> Enhanced Influence Model</a><ul>
<li class="chapter" data-level="5.1" data-path="enhanced-influence-model.html"><a href="enhanced-influence-model.html#graph-diffusion-kernel"><i class="fa fa-check"></i><b>5.1</b> Graph Diffusion Kernel</a></li>
<li class="chapter" data-level="5.2" data-path="enhanced-influence-model.html"><a href="enhanced-influence-model.html#enhanced-influence-model-1"><i class="fa fa-check"></i><b>5.2</b> Enhanced Influence Model</a></li>
<li class="chapter" data-level="5.3" data-path="enhanced-influence-model.html"><a href="enhanced-influence-model.html#determine-value-for-cold-edge-threshold-delta"><i class="fa fa-check"></i><b>5.3</b> Determine Value for Cold-Edge Threshold <span class="math inline">\(\delta\)</span></a></li>
<li class="chapter" data-level="5.4" data-path="enhanced-influence-model.html"><a href="enhanced-influence-model.html#alternative-situation"><i class="fa fa-check"></i><b>5.4</b> Alternative Situation</a></li>
<li class="chapter" data-level="5.5" data-path="enhanced-influence-model.html"><a href="enhanced-influence-model.html#code"><i class="fa fa-check"></i><b>5.5</b> Code</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="discussion-and-future-work.html"><a href="discussion-and-future-work.html"><i class="fa fa-check"></i><b>6</b> Discussion and Future Work</a><ul>
<li class="chapter" data-level="6.1" data-path="discussion-and-future-work.html"><a href="discussion-and-future-work.html#what-did-we-achieve"><i class="fa fa-check"></i><b>6.1</b> What did we achieve?</a></li>
<li class="chapter" data-level="6.2" data-path="discussion-and-future-work.html"><a href="discussion-and-future-work.html#limitations"><i class="fa fa-check"></i><b>6.2</b> Limitations</a></li>
<li class="chapter" data-level="6.3" data-path="discussion-and-future-work.html"><a href="discussion-and-future-work.html#future-work"><i class="fa fa-check"></i><b>6.3</b> Future Work</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>7</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">MATH 479 Network Science Project</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="enhanced-influence-model" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Enhanced Influence Model</h1>
<p>After constructing a network using our simulated data, we moved on to use the Enhanced Influence Model to identify significant subnetworks within the graph that shed light on genetic causal pathways. We will first introduce the graph diffusion kernel as the first step to measure the extent of mutual interaction between genes in our network.</p>
<div id="graph-diffusion-kernel" class="section level2">
<h2><span class="header-section-number">5.1</span> Graph Diffusion Kernel</h2>
<p>Here we imagine that the importance of nodes flows along edges. Initially, query nodes are selected to serve as sources where fluid is pumped in at a constant rate which then diffuses through the network. In addition, we assume that fluid leaks out of each node at a constant rate <span class="math inline">\(\gamma\)</span>. Intuitively, large <span class="math inline">\(\gamma\)</span> means faster rate of loss and therefore shorter diffusive paths. Essentially, the paths reflects diffusion of random walkers from a source. At equilibrium, there is no net flow in the entire network. The more interaction a node has with the query nodes, the more fluid it will contain at the equilibrium (Qi et al., 2008). Following the procedure in previous literature (Vandin, Upfal, and Raphael, 2011), we set <span class="math inline">\(\gamma = 5\)</span>, which is approximately the average degree of a node in our network.</p>
<p>Mathematically,let <span class="math inline">\(u(t)\)</span> denotes the unit step function where <span class="math display">\[u(t) = \begin{cases}0 \;\text{ if }\;t&lt;0\\1 \;\text{ if }\;t&gt;0 \end{cases},\]</span>
and <span class="math inline">\(b_i = 1\)</span> if i is the source node of interest while <span class="math inline">\(b_i = 0\)</span> otherwise.</p>
<p>The amount of fluid contained by node <span class="math inline">\(i\)</span> at time t is <span class="math inline">\(x_i(t)\)</span> is governed by the flow in and out of it,</p>
<p><span class="math display">\[\frac{d}{dt}x_i(t) = \sum_j A_{ij}x_j(t)-\sum_jA_{ji}x_i(t)-\gamma x_i(t)\]</span>
As <span class="math inline">\(t\to \infty\)</span>, the fluid distribution approaches to its equilibrium distribution at which the influence of node <span class="math inline">\(i\)</span> to other nodes is measured by <span class="math display">\[L_\gamma^{-1}\vec{b_i}\]</span>
where <span class="math inline">\(\vec{b_i}\)</span> is the standard basis with 1 at its <span class="math inline">\(i^{th}\)</span> entry. Here, <span class="math inline">\(L_\gamma = S-A+\gamma I\)</span> is the shifted graph Laplacian by <span class="math inline">\(\gamma\)</span> where <span class="math inline">\(S\)</span> is the diagonal matrix whose diagonal entries are the corresponding degrees. In practice, we normalize A symmetrically by <span class="math inline">\(S^{-\frac{1}{2}}AS^{-\frac{1}{2}}\)</span> such that the edge weights between two nodes are normalized by the degrees and replace <span class="math inline">\(S\)</span> correspondingly to obtain the Laplacian (Qi et al. 2008).</p>
<p>Here, the <span class="math inline">\(ij\)</span> entry of <span class="math inline">\(L_\gamma^{-1}\)</span> gives the influence of <span class="math inline">\(j\)</span> on <span class="math inline">\(i\)</span>, which may or may not equal to the <span class="math inline">\(ji\)</span> entry of <span class="math inline">\(L_\gamma^{-1}\)</span>. To better interpret the result, we define the mutual influence between i and j to be <span class="math inline">\(\tilde{i}(i,j) = \min(L_\gamma^{-1}[i,j],L_\gamma^{-1}[j,i])\)</span>. (Vandin, Upfal, and Raphael, 2011)</p>
</div>
<div id="enhanced-influence-model-1" class="section level2">
<h2><span class="header-section-number">5.2</span> Enhanced Influence Model</h2>
<p>Enhanced Influence Model is an efficient algorithm to identify a significantly mutated subnetwork with respect to gene interactions by incorporating considerations of both the topology of the network and significance of individual genes with respect to the disease (Vandin, Upfal, and Raphael, 2011). Below are the key procedures involved in the Enhanced Influence Model:</p>
<ul>
<li><p>Recall from Chapter 4, marginal regression gives a p-value, <span class="math inline">\(p_i\)</span> for each gene <span class="math inline">\(i\)</span>. We then assign a score <span class="math inline">\(\sigma_i\)</span> to a gene <span class="math inline">\(i\)</span> where <span class="math inline">\(\sigma_i = -2\log(p_i)\)</span>.</p></li>
<li><p>For each pair of genes (<span class="math inline">\(i\)</span>,<span class="math inline">\(j\)</span>), their mutual interaction is measured by <span class="math inline">\(w(i,j) = \max(\sigma_i,\sigma_j)*\tilde{i}(i,j)\)</span> (Vandin et al., 2011).</p></li>
<li><p>Next, we remove all edges with weights smaller than a threshold <span class="math inline">\(\delta\)</span> to obtain a subgraph. In Section 5.3, we shall exhibit a method to determine this cold-edge threshold <span class="math inline">\(\delta\)</span>. The connected components remained are candidates for genetic pathways.</p></li>
</ul>
<p>The code below shows how we construct the enhanced influence model from scratch.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="enhanced-influence-model.html#cb26-1"></a><span class="co"># normalize A</span></span>
<span id="cb26-2"><a href="enhanced-influence-model.html#cb26-2"></a>A_normalized &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>,<span class="kw">nrow</span>(A),<span class="kw">nrow</span>(A))</span>
<span id="cb26-3"><a href="enhanced-influence-model.html#cb26-3"></a></span>
<span id="cb26-4"><a href="enhanced-influence-model.html#cb26-4"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>(<span class="kw">nrow</span>(A)<span class="op">-</span><span class="dv">1</span>)){</span>
<span id="cb26-5"><a href="enhanced-influence-model.html#cb26-5"></a>  <span class="cf">for</span>(j <span class="cf">in</span> i<span class="op">:</span><span class="kw">nrow</span>(A)){</span>
<span id="cb26-6"><a href="enhanced-influence-model.html#cb26-6"></a>    <span class="cf">if</span>(deg[i]<span class="op">!=</span><span class="dv">0</span> <span class="op">&amp;&amp;</span><span class="st"> </span>deg[j]<span class="op">!=</span><span class="dv">0</span>){</span>
<span id="cb26-7"><a href="enhanced-influence-model.html#cb26-7"></a>      A_normalized[i,j] &lt;-<span class="st"> </span>A[i,j]<span class="op">/</span><span class="kw">sqrt</span>(deg[i]<span class="op">*</span>deg[j])</span>
<span id="cb26-8"><a href="enhanced-influence-model.html#cb26-8"></a>    }</span>
<span id="cb26-9"><a href="enhanced-influence-model.html#cb26-9"></a>  }</span>
<span id="cb26-10"><a href="enhanced-influence-model.html#cb26-10"></a>}</span>
<span id="cb26-11"><a href="enhanced-influence-model.html#cb26-11"></a></span>
<span id="cb26-12"><a href="enhanced-influence-model.html#cb26-12"></a>A_normalized &lt;-<span class="st"> </span><span class="kw">forceSymmetric</span>(A_normalized,<span class="st">&quot;U&quot;</span>)</span>
<span id="cb26-13"><a href="enhanced-influence-model.html#cb26-13"></a>A_normalized &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(A_normalized)</span>
<span id="cb26-14"><a href="enhanced-influence-model.html#cb26-14"></a></span>
<span id="cb26-15"><a href="enhanced-influence-model.html#cb26-15"></a><span class="co"># normalize the corresponding degree matrix S</span></span>
<span id="cb26-16"><a href="enhanced-influence-model.html#cb26-16"></a>S_normalized &lt;-<span class="st"> </span><span class="kw">diag</span>(<span class="kw">rowSums</span>(A_normalized))</span>
<span id="cb26-17"><a href="enhanced-influence-model.html#cb26-17"></a></span>
<span id="cb26-18"><a href="enhanced-influence-model.html#cb26-18"></a>gamma &lt;-<span class="st"> </span><span class="dv">5</span></span>
<span id="cb26-19"><a href="enhanced-influence-model.html#cb26-19"></a>L_gamma &lt;-<span class="st"> </span>S_normalized<span class="op">+</span><span class="kw">diag</span>(gamma,<span class="kw">nrow</span>(A))<span class="op">-</span>A_normalized</span>
<span id="cb26-20"><a href="enhanced-influence-model.html#cb26-20"></a>inv_L_gamma &lt;-<span class="st"> </span><span class="kw">solve</span>(L_gamma)</span></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="enhanced-influence-model.html#cb27-1"></a><span class="co"># compute symmetric importance score using Enhanced Influence Model</span></span>
<span id="cb27-2"><a href="enhanced-influence-model.html#cb27-2"></a>W &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="kw">nrow</span>(A),<span class="kw">ncol</span>(A))</span>
<span id="cb27-3"><a href="enhanced-influence-model.html#cb27-3"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>(<span class="kw">nrow</span>(A)<span class="op">-</span><span class="dv">1</span>)){</span>
<span id="cb27-4"><a href="enhanced-influence-model.html#cb27-4"></a>  <span class="cf">for</span>(j <span class="cf">in</span> (i<span class="op">+</span><span class="dv">1</span>)<span class="op">:</span><span class="kw">nrow</span>(A)){</span>
<span id="cb27-5"><a href="enhanced-influence-model.html#cb27-5"></a>    W[i,j] &lt;-<span class="st"> </span><span class="kw">min</span>(inv_L_gamma[i,j],inv_L_gamma[j,i])<span class="op">*</span><span class="kw">max</span>(mutations_score[i],mutations_score[j])</span>
<span id="cb27-6"><a href="enhanced-influence-model.html#cb27-6"></a>  }</span>
<span id="cb27-7"><a href="enhanced-influence-model.html#cb27-7"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="enhanced-influence-model.html#cb28-1"></a><span class="co"># helper function to remove cold_edge</span></span>
<span id="cb28-2"><a href="enhanced-influence-model.html#cb28-2"></a>removeColdEdge &lt;-<span class="st"> </span><span class="cf">function</span>(mat,W,delta){</span>
<span id="cb28-3"><a href="enhanced-influence-model.html#cb28-3"></a>  hotnet &lt;-<span class="st"> </span>mat</span>
<span id="cb28-4"><a href="enhanced-influence-model.html#cb28-4"></a></span>
<span id="cb28-5"><a href="enhanced-influence-model.html#cb28-5"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>(<span class="kw">nrow</span>(A)<span class="op">-</span><span class="dv">1</span>)){</span>
<span id="cb28-6"><a href="enhanced-influence-model.html#cb28-6"></a>    <span class="cf">for</span>(j <span class="cf">in</span> (i<span class="op">+</span><span class="dv">1</span>)<span class="op">:</span><span class="kw">nrow</span>(A)){</span>
<span id="cb28-7"><a href="enhanced-influence-model.html#cb28-7"></a>      <span class="cf">if</span>(W[i,j]<span class="op">&lt;</span>delta){</span>
<span id="cb28-8"><a href="enhanced-influence-model.html#cb28-8"></a>        hotnet[i,j] =<span class="st"> </span><span class="dv">0</span></span>
<span id="cb28-9"><a href="enhanced-influence-model.html#cb28-9"></a>      }</span>
<span id="cb28-10"><a href="enhanced-influence-model.html#cb28-10"></a>    }</span>
<span id="cb28-11"><a href="enhanced-influence-model.html#cb28-11"></a>  }</span>
<span id="cb28-12"><a href="enhanced-influence-model.html#cb28-12"></a>  <span class="kw">return</span>(hotnet)</span>
<span id="cb28-13"><a href="enhanced-influence-model.html#cb28-13"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="enhanced-influence-model.html#cb29-1"></a>Hotnet &lt;-<span class="st"> </span><span class="kw">removeColdEdge</span>(A_normalized,W,<span class="fl">0.04</span>)</span>
<span id="cb29-2"><a href="enhanced-influence-model.html#cb29-2"></a>Hotnet &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">forceSymmetric</span>(Hotnet,<span class="st">&quot;U&quot;</span>))</span>
<span id="cb29-3"><a href="enhanced-influence-model.html#cb29-3"></a><span class="kw">colnames</span>(Hotnet) &lt;-<span class="st"> </span>position</span>
<span id="cb29-4"><a href="enhanced-influence-model.html#cb29-4"></a><span class="kw">rownames</span>(Hotnet) &lt;-<span class="st"> </span>position</span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="enhanced-influence-model.html#cb30-1"></a>HotEdges &lt;-<span class="st"> </span><span class="kw">melt</span>(Hotnet) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb30-2"><a href="enhanced-influence-model.html#cb30-2"></a><span class="st">  </span><span class="kw">rename</span>(<span class="dt">Source =</span> Var1, <span class="dt">Target =</span> Var2, <span class="dt">Weight =</span> value) <span class="op">%&gt;%</span></span>
<span id="cb30-3"><a href="enhanced-influence-model.html#cb30-3"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Type =</span> <span class="st">&quot;Undirected&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb30-4"><a href="enhanced-influence-model.html#cb30-4"></a><span class="st">  </span><span class="kw">select</span>(Source, Target, Type,Weight) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb30-5"><a href="enhanced-influence-model.html#cb30-5"></a><span class="st">  </span><span class="kw">filter</span>(Weight <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span></span>
<span id="cb30-6"><a href="enhanced-influence-model.html#cb30-6"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>Weight)</span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="enhanced-influence-model.html#cb31-1"></a>edges.copy &lt;-<span class="st"> </span>edges</span>
<span id="cb31-2"><a href="enhanced-influence-model.html#cb31-2"></a>edges.copy &lt;-<span class="st"> </span>edges.copy <span class="op">%&gt;%</span></span>
<span id="cb31-3"><a href="enhanced-influence-model.html#cb31-3"></a><span class="st">  </span><span class="kw">select</span>(Source,Target)</span>
<span id="cb31-4"><a href="enhanced-influence-model.html#cb31-4"></a><span class="kw">colnames</span>(edges.copy) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;from&quot;</span>,<span class="st">&quot;to&quot;</span>)</span>
<span id="cb31-5"><a href="enhanced-influence-model.html#cb31-5"></a></span>
<span id="cb31-6"><a href="enhanced-influence-model.html#cb31-6"></a>g &lt;-<span class="st"> </span><span class="kw">graph_from_data_frame</span>(edges.copy, <span class="dt">directed=</span><span class="ot">FALSE</span>, <span class="dt">vertices=</span>nodes)</span>
<span id="cb31-7"><a href="enhanced-influence-model.html#cb31-7"></a></span>
<span id="cb31-8"><a href="enhanced-influence-model.html#cb31-8"></a><span class="co"># get a list of connected components</span></span>
<span id="cb31-9"><a href="enhanced-influence-model.html#cb31-9"></a>clu &lt;-<span class="st"> </span><span class="kw">components</span>(g)</span>
<span id="cb31-10"><a href="enhanced-influence-model.html#cb31-10"></a>res &lt;-<span class="st"> </span><span class="kw">groups</span>(clu)</span></code></pre></div>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="enhanced-influence-model.html#cb32-1"></a>getNumComp &lt;-<span class="st"> </span><span class="cf">function</span>(edges,nodes,s){</span>
<span id="cb32-2"><a href="enhanced-influence-model.html#cb32-2"></a>  edges.copy =<span class="st"> </span>edges</span>
<span id="cb32-3"><a href="enhanced-influence-model.html#cb32-3"></a>  edges.copy =<span class="st"> </span>edges.copy<span class="op">%&gt;%</span><span class="kw">select</span>(Source,Target)</span>
<span id="cb32-4"><a href="enhanced-influence-model.html#cb32-4"></a>  <span class="kw">colnames</span>(edges.copy) =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;from&quot;</span>,<span class="st">&quot;to&quot;</span>)</span>
<span id="cb32-5"><a href="enhanced-influence-model.html#cb32-5"></a></span>
<span id="cb32-6"><a href="enhanced-influence-model.html#cb32-6"></a>  g &lt;-<span class="st"> </span><span class="kw">graph_from_data_frame</span>(edges.copy, <span class="dt">directed=</span><span class="ot">FALSE</span>, <span class="dt">vertices=</span>nodes)</span>
<span id="cb32-7"><a href="enhanced-influence-model.html#cb32-7"></a>  clu &lt;-<span class="st"> </span><span class="kw">components</span>(g)</span>
<span id="cb32-8"><a href="enhanced-influence-model.html#cb32-8"></a>  res &lt;-<span class="st"> </span><span class="kw">groups</span>(clu)</span>
<span id="cb32-9"><a href="enhanced-influence-model.html#cb32-9"></a>  count &lt;-<span class="st"> </span><span class="dv">0</span></span>
<span id="cb32-10"><a href="enhanced-influence-model.html#cb32-10"></a>  </span>
<span id="cb32-11"><a href="enhanced-influence-model.html#cb32-11"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(res)){</span>
<span id="cb32-12"><a href="enhanced-influence-model.html#cb32-12"></a>    <span class="cf">if</span>(<span class="kw">length</span>(res[[i]]) <span class="op">&gt;=</span><span class="st"> </span>s){</span>
<span id="cb32-13"><a href="enhanced-influence-model.html#cb32-13"></a>      count =<span class="st"> </span>count<span class="op">+</span><span class="dv">1</span></span>
<span id="cb32-14"><a href="enhanced-influence-model.html#cb32-14"></a>    }</span>
<span id="cb32-15"><a href="enhanced-influence-model.html#cb32-15"></a>  }</span>
<span id="cb32-16"><a href="enhanced-influence-model.html#cb32-16"></a>  <span class="kw">return</span>(count)</span>
<span id="cb32-17"><a href="enhanced-influence-model.html#cb32-17"></a>}</span></code></pre></div>
</div>
<div id="determine-value-for-cold-edge-threshold-delta" class="section level2">
<h2><span class="header-section-number">5.3</span> Determine Value for Cold-Edge Threshold <span class="math inline">\(\delta\)</span></h2>
<p>In this section, we illustrate a way to determine the value for <span class="math inline">\(\delta\)</span>. We start with simulating 50 datasets under <span class="math inline">\(H_0^P\)</span> by permuting observed disease status. (Vandin et. al, 2012) Then, for a small value of <span class="math inline">\(s\)</span>, where often <span class="math inline">\(s \in \{3,4,5\}\)</span>, we choose the first <span class="math inline">\(\delta\)</span> that gives the largest number of subnetworks of size at least <span class="math inline">\(s\)</span>. Given the size of our network, we use <span class="math inline">\(s = 3\)</span>. Table 5.1 report the number of connected components of size at least 3 as a function of <span class="math inline">\(\delta\)</span>.</p>
<table class="table lightable-paper lightable-striped" style='margin-left: auto; margin-right: auto; font-family: "Arial Narrow", arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;'>
<tbody>
<tr>
<td style="text-align:left;">
delta value
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.06
</td>
<td style="text-align:right;">
0.07
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.09
</td>
</tr>
<tr>
<td style="text-align:left;">
Num.Components, Size&gt;3
</td>
<td style="text-align:right;">
100.30
</td>
<td style="text-align:right;">
118.10
</td>
<td style="text-align:right;">
50.20
</td>
<td style="text-align:right;">
10.50
</td>
<td style="text-align:right;">
2.40
</td>
<td style="text-align:right;">
2.40
</td>
<td style="text-align:right;">
2.40
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
</tbody>
</table>

<p><span class="math display">\[\text{Table 5.1: Number of Components of Size at Least 3 as a Function of} \;\delta\]</span></p>
<p>According to Vandin (2012), an empirical choice is using a first <span class="math inline">\(\delta\)</span> that gives the maximum number of connected components of size at least 3, which gives <span class="math inline">\(\delta = 0.02\)</span> in our case. Since increasing <span class="math inline">\(\delta\)</span> gives us more conservative criteria, we also consider the following values: <span class="math inline">\(1.5\delta = 0.03, 2\delta = 0.04, 3\delta = 0.06, 4\delta = 0.08\)</span>. In fact, if <span class="math inline">\(\delta_1 &gt;\delta_2\)</span>, then the connected components of size at least 3 obtained using <span class="math inline">\(\delta_2\)</span> will be subnetworks of those obtained using <span class="math inline">\(\delta_1\)</span>.</p>
<p>We applied the above <span class="math inline">\(\delta\)</span> values to filter the edges and obtained 5 graphs using the graph presented in Chapter 4. Intuitively, as we raise the value for <span class="math inline">\(\delta\)</span>, there will be more isolated nodes and the size of major component will decrease as it gets decomposed into smaller parts. We expect the the true causal pathways will be among the connected components after applying the Enhanced Influence Model and filtering the edges. Figures 5.1 (a) - (e) shows their corresponding major components, which contained the 4 genetic causal pathways listed in Chapter 4.</p>
<center>
<img src="delta_major.png" style="width:50.0%" />
</center>
<p><span class="math display">\[\text{Figure 5.1(a): Major component of the network of mutated genes when delta = 0.02}\]</span>
<span class="math display">\[\text{Nodes Sized by Closeness Centrality}\]</span></p>
<center>
<img src="1.5delta_major.png" style="width:50.0%" />
</center>
<p><span class="math display">\[\text{Figure 5.1(b): Major component of the network of mutated genes when delta = 0.03}\]</span></p>
<center>
<img src="2delta_major.png" style="width:50.0%" />
</center>
<p><span class="math display">\[\text{Figure 5.1(c): Major component of the network of mutated genes when delta = 0.04}\]</span></p>
<center>
<img src="3Delta_major.png" style="width:50.0%" />
</center>
<p><span class="math display">\[\text{Figure 5.1(d): Major component of the network of mutated genes when delta = 0.06}\]</span></p>
<center>
<img src="4delta.png" style="width:40.0%" />
</center>
<p><span class="math display">\[\text{Figure 5.1(e): Major component of the network of mutated genes when delta = 0.0}\]</span></p>
<p>We found that at <span class="math inline">\(\delta = 0.04\)</span> and <span class="math inline">\(\delta= 0.06\)</span>, the major component both contains precisely the 12 SNP locations that were part of at least one pathway interaction combination. To determine which one of the two is better, we go back to the four original pathways and evaluate their quality. It seems that the major component more accurately describes the truth when <span class="math inline">\(\delta = 0.04\)</span>, as presented below.</p>
<center>
<img src="mod1_hotedge0.04_closeness.png" style="width:50.0%" />
</center>
<p><span class="math display">\[\text{Figure 5.2: Modularity Class Breakdown within the Major Component when delta = 0.04}\]</span></p>
</div>
<div id="alternative-situation" class="section level2">
<h2><span class="header-section-number">5.4</span> Alternative Situation</h2>
<p>In the previous section, we find that our method is confused about which modularity class a certain node belongs to. We hypothesize that this may be due to that the node makes up more than one combination of pathway interactions. We wonder if the problem will arise when each mutation only contributes to one pathway combination (denoted as Scenario B). If the problem universally happens, we need to analyze its cause.</p>
<p>We repeated the process, with a twist in the construction of the original dataset. We simulated a dataset where no two pathways share a SNP location. Random simulation gives the following 4 causal genetic pathways:</p>
<ol style="list-style-type: decimal">
<li><p>X2608, X182, X1443, X624, X2422</p></li>
<li><p>X2454, X1099, X350, X2548, X883</p></li>
<li><p>X31, X2725, X608, X471</p></li>
<li><p>X815, X2108, X2494, X2812, X1245</p></li>
</ol>
<p>Figure 5.3 shows the full network of SNP locations where at least one mutation occurs where the purple part is the largest connected component.</p>
<p><img src="mod2_full.png" />
<span class="math display">\[\text{Figure 5.3: Figure Full Network of Mutated Genes under Scenario B}\]</span>
 </p>
<table class="table lightable-paper lightable-striped" style='margin-left: auto; margin-right: auto; font-family: "Arial Narrow", arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;'>
<tbody>
<tr>
<td style="text-align:left;">
delta value
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
0.02
</td>
<td style="text-align:right;">
0.03
</td>
<td style="text-align:right;">
0.04
</td>
<td style="text-align:right;">
0.05
</td>
<td style="text-align:right;">
0.06
</td>
<td style="text-align:right;">
0.07
</td>
<td style="text-align:right;">
0.08
</td>
<td style="text-align:right;">
0.09
</td>
</tr>
<tr>
<td style="text-align:left;">
Num.Components, Size&gt;3
</td>
<td style="text-align:right;">
100.30
</td>
<td style="text-align:right;">
118.10
</td>
<td style="text-align:right;">
50.20
</td>
<td style="text-align:right;">
10.50
</td>
<td style="text-align:right;">
2.40
</td>
<td style="text-align:right;">
2.40
</td>
<td style="text-align:right;">
2.40
</td>
<td style="text-align:right;">
0.00
</td>
<td style="text-align:right;">
0.00
</td>
</tr>
</tbody>
</table>

<p><span class="math display">\[\text{Table 5.2: Number of Components of Size at Least 3 as a Function of} \;\delta \;\text{under Scenario B}\]</span>
The empirical choice <span class="math inline">\(\delta&#39; = 0.04\)</span> gives the first mamimum average number of connected components of size at least 3. Experimenting with <span class="math inline">\(\delta = 1.5\delta&#39;\)</span> and <span class="math inline">\(\delta = 2\delta&#39;\)</span> shows that <span class="math inline">\(\delta = 2\delta&#39;=0.08\)</span> gives a perfect fit.</p>
<p>Figure 5.4 shows the filtered network obtained from Figure 5.3 using <span class="math inline">\(\delta = 0.08\)</span>. Observe that there is no single major connected components. However, all four pathways exist as independent connected components that are colored in Figure 5.4. Figure 5.5 shows the the filtered graph for the four components. All 4 pathwaysa re identified, which supplies proof for the validity of our method.</p>
<p><img src="mod2_HotEdges0.08.png" /></p>
<p><span class="math display">\[\text{Figure 5.4: Resulted Network using}\; \delta=0.08 \;\text{under Scenario B}\]</span></p>
<center>
<img src="mod2_hotedge_2delta_0.08_major.png" style="width:50.0%" />
</center>
<p><span class="math display">\[\text{Figure 5.5: All 4 Pathways Identified under Scenario B}\]</span></p>
</div>
<div id="code" class="section level2">
<h2><span class="header-section-number">5.5</span> Code</h2>
<p>Below is the code that supports our work in this chapter.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="enhanced-influence-model.html#cb33-1"></a>disease_status_mat &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="ot">NA</span>,<span class="dv">50</span>,<span class="dv">500</span>)</span>
<span id="cb33-2"><a href="enhanced-influence-model.html#cb33-2"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">50</span>){</span>
<span id="cb33-3"><a href="enhanced-influence-model.html#cb33-3"></a>  disease_status_mat[i,] &lt;-<span class="st"> </span><span class="kw">permute</span>(disease_status_vector)</span>
<span id="cb33-4"><a href="enhanced-influence-model.html#cb33-4"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="enhanced-influence-model.html#cb34-1"></a>pVal.lst &lt;-<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb34-2"><a href="enhanced-influence-model.html#cb34-2"></a><span class="cf">for</span>(k <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">50</span>){</span>
<span id="cb34-3"><a href="enhanced-influence-model.html#cb34-3"></a>  res &lt;-<span class="st"> </span><span class="kw">c</span>()</span>
<span id="cb34-4"><a href="enhanced-influence-model.html#cb34-4"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(position)){</span>
<span id="cb34-5"><a href="enhanced-influence-model.html#cb34-5"></a>    outcome &lt;-<span class="st"> &quot;disease_status_mat[k,]&quot;</span></span>
<span id="cb34-6"><a href="enhanced-influence-model.html#cb34-6"></a>    variable &lt;-<span class="st"> </span>position[i]</span>
<span id="cb34-7"><a href="enhanced-influence-model.html#cb34-7"></a>    f &lt;-<span class="st"> </span><span class="kw">as.formula</span>(<span class="kw">paste</span>(outcome, </span>
<span id="cb34-8"><a href="enhanced-influence-model.html#cb34-8"></a>        <span class="kw">paste</span>(variable), </span>
<span id="cb34-9"><a href="enhanced-influence-model.html#cb34-9"></a>        <span class="dt">sep =</span> <span class="st">&quot; ~ &quot;</span>))</span>
<span id="cb34-10"><a href="enhanced-influence-model.html#cb34-10"></a>    model &lt;-<span class="st"> </span><span class="kw">lm</span>(f, <span class="dt">data =</span> sub_data)</span>
<span id="cb34-11"><a href="enhanced-influence-model.html#cb34-11"></a>    res &lt;-<span class="st"> </span><span class="kw">c</span>(res,<span class="kw">summary</span>(model)<span class="op">$</span>coefficient[,<span class="dv">4</span>][<span class="dv">2</span>])</span>
<span id="cb34-12"><a href="enhanced-influence-model.html#cb34-12"></a>    pVal.lst[[k]] =<span class="st"> </span>res</span>
<span id="cb34-13"><a href="enhanced-influence-model.html#cb34-13"></a>  }</span>
<span id="cb34-14"><a href="enhanced-influence-model.html#cb34-14"></a>  </span>
<span id="cb34-15"><a href="enhanced-influence-model.html#cb34-15"></a>}</span>
<span id="cb34-16"><a href="enhanced-influence-model.html#cb34-16"></a>score.lst =<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb34-17"><a href="enhanced-influence-model.html#cb34-17"></a><span class="cf">for</span>(k <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">50</span>){</span>
<span id="cb34-18"><a href="enhanced-influence-model.html#cb34-18"></a>  score.lst[[k]] =<span class="st"> </span><span class="dv">-2</span><span class="op">*</span><span class="kw">log</span>(pVal.lst[[k]])</span>
<span id="cb34-19"><a href="enhanced-influence-model.html#cb34-19"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="enhanced-influence-model.html#cb35-1"></a>W.lst =<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb35-2"><a href="enhanced-influence-model.html#cb35-2"></a></span>
<span id="cb35-3"><a href="enhanced-influence-model.html#cb35-3"></a><span class="cf">for</span>(k <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">50</span>){</span>
<span id="cb35-4"><a href="enhanced-influence-model.html#cb35-4"></a>  W.lst[[k]] =<span class="st"> </span><span class="kw">matrix</span>(<span class="ot">NA</span>, <span class="kw">nrow</span>(A),<span class="kw">ncol</span>(A))</span>
<span id="cb35-5"><a href="enhanced-influence-model.html#cb35-5"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span>(<span class="kw">nrow</span>(A)<span class="op">-</span><span class="dv">1</span>)){</span>
<span id="cb35-6"><a href="enhanced-influence-model.html#cb35-6"></a>    <span class="cf">for</span>(j <span class="cf">in</span> (i<span class="op">+</span><span class="dv">1</span>)<span class="op">:</span><span class="kw">nrow</span>(A)){</span>
<span id="cb35-7"><a href="enhanced-influence-model.html#cb35-7"></a>      W.lst[[k]][i,j] =<span class="st"> </span><span class="kw">min</span>(inv_L_gamma[i,j],inv_L_gamma[j,i])<span class="op">*</span><span class="kw">max</span>(score.lst[[k]][i],score.lst[[k]][j])</span>
<span id="cb35-8"><a href="enhanced-influence-model.html#cb35-8"></a>    }</span>
<span id="cb35-9"><a href="enhanced-influence-model.html#cb35-9"></a>  }</span>
<span id="cb35-10"><a href="enhanced-influence-model.html#cb35-10"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="enhanced-influence-model.html#cb36-1"></a>aveNumComp_s &lt;-<span class="st"> </span><span class="cf">function</span>(delta, <span class="dt">s=</span><span class="dv">3</span>){</span>
<span id="cb36-2"><a href="enhanced-influence-model.html#cb36-2"></a>  <span class="co"># list of adjacency matrices for HotNet for 50 datasets under the null hypothesis</span></span>
<span id="cb36-3"><a href="enhanced-influence-model.html#cb36-3"></a>  <span class="co"># under the given threshold delta</span></span>
<span id="cb36-4"><a href="enhanced-influence-model.html#cb36-4"></a>  Hotnet.lst =<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb36-5"><a href="enhanced-influence-model.html#cb36-5"></a>  <span class="cf">for</span>(k <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>){</span>
<span id="cb36-6"><a href="enhanced-influence-model.html#cb36-6"></a>    Hotnet.lst[[k]] =<span class="st"> </span><span class="kw">removeColdEdge</span>(A_normalized,W.lst[[k]],delta)</span>
<span id="cb36-7"><a href="enhanced-influence-model.html#cb36-7"></a>    Hotnet.lst[[k]] =<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">forceSymmetric</span>(Hotnet.lst[[k]],<span class="st">&quot;U&quot;</span>))</span>
<span id="cb36-8"><a href="enhanced-influence-model.html#cb36-8"></a>  }</span>
<span id="cb36-9"><a href="enhanced-influence-model.html#cb36-9"></a></span>
<span id="cb36-10"><a href="enhanced-influence-model.html#cb36-10"></a>  HotEdges.lst =<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb36-11"><a href="enhanced-influence-model.html#cb36-11"></a>  <span class="cf">for</span>(k <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>){</span>
<span id="cb36-12"><a href="enhanced-influence-model.html#cb36-12"></a>    <span class="kw">colnames</span>(Hotnet.lst[[k]]) =<span class="st"> </span>position</span>
<span id="cb36-13"><a href="enhanced-influence-model.html#cb36-13"></a>    <span class="kw">rownames</span>(Hotnet.lst[[k]]) =<span class="st"> </span>position</span>
<span id="cb36-14"><a href="enhanced-influence-model.html#cb36-14"></a>    HotEdges.lst[[k]]=<span class="st"> </span><span class="kw">melt</span>(Hotnet.lst[[k]]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb36-15"><a href="enhanced-influence-model.html#cb36-15"></a><span class="st">      </span><span class="kw">rename</span>(<span class="dt">Source =</span> Var1, <span class="dt">Target =</span> Var2, <span class="dt">Weight =</span> value)<span class="op">%&gt;%</span></span>
<span id="cb36-16"><a href="enhanced-influence-model.html#cb36-16"></a><span class="st">      </span><span class="kw">mutate</span>(<span class="dt">Type =</span> <span class="st">&quot;Undirected&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb36-17"><a href="enhanced-influence-model.html#cb36-17"></a><span class="st">      </span><span class="kw">select</span>(Source, Target, Type,Weight) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb36-18"><a href="enhanced-influence-model.html#cb36-18"></a><span class="st">      </span><span class="kw">filter</span>(Weight <span class="op">!=</span><span class="st"> </span><span class="dv">0</span>)<span class="op">%&gt;%</span><span class="kw">select</span>(<span class="op">-</span>Weight)</span>
<span id="cb36-19"><a href="enhanced-influence-model.html#cb36-19"></a>  }</span>
<span id="cb36-20"><a href="enhanced-influence-model.html#cb36-20"></a></span>
<span id="cb36-21"><a href="enhanced-influence-model.html#cb36-21"></a>  numComp_s =<span class="st"> </span><span class="kw">rep</span>(<span class="dv">0</span>,<span class="dv">10</span>)</span>
<span id="cb36-22"><a href="enhanced-influence-model.html#cb36-22"></a>  <span class="cf">for</span>(k <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>){</span>
<span id="cb36-23"><a href="enhanced-influence-model.html#cb36-23"></a>    numComp_s[k] =<span class="st"> </span><span class="kw">getNumComp</span>(HotEdges.lst[[k]],nodes,s)</span>
<span id="cb36-24"><a href="enhanced-influence-model.html#cb36-24"></a>    </span>
<span id="cb36-25"><a href="enhanced-influence-model.html#cb36-25"></a>  }</span>
<span id="cb36-26"><a href="enhanced-influence-model.html#cb36-26"></a>  <span class="kw">return</span>(<span class="kw">mean</span>(numComp_s))</span>
<span id="cb36-27"><a href="enhanced-influence-model.html#cb36-27"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="enhanced-influence-model.html#cb37-1"></a>x &lt;-<span class="st"> </span><span class="kw">seq</span>(<span class="fl">0.01</span>,<span class="fl">0.09</span>, <span class="fl">0.01</span>)</span>
<span id="cb37-2"><a href="enhanced-influence-model.html#cb37-2"></a><span class="kw">sapply</span>(x, aveNumComp_s,<span class="dt">s=</span><span class="dv">3</span>)</span></code></pre></div>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="enhanced-influence-model.html#cb38-1"></a>x &lt;-<span class="st"> </span><span class="kw">seq</span>(<span class="fl">0.01</span>,<span class="fl">0.09</span>, <span class="fl">0.01</span>)</span>
<span id="cb38-2"><a href="enhanced-influence-model.html#cb38-2"></a><span class="kw">sapply</span>(x, aveNumComp_s,<span class="dt">s=</span><span class="dv">2</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="network-construction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="discussion-and-future-work.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "chapter",
"scroll_highlight": true
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
